src/strings-in-block/class-compose.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace WPML\PB\Gutenberg\StringsInBlock;
+
+class Compose implements StringsInBlock {
+
+	/** @var StringsInBlock[] $parsers */
+	private $parsers = [];
+
+	/**
+	 * @param StringsInBlock[] $parsers
+	 */
+	public function __construct( array $parsers ) {
+		$this->parsers = $parsers;
+	}
+
+	/**
+	 * @param \WP_Block_Parser_Block $block
+	 *
+	 * @return array
+	 */
+	public function find( \WP_Block_Parser_Block $block ) {
+		$strings = [];
+
+		foreach ( $this->parsers as $parser ) {
+			$strings = array_merge( $strings, $parser->find( $block ) );
+		}
```

Comment:
@strategio If $this->parsers is made as a collection in the constructor then you can do the following. But I'm not sure if it's better or not.

```php
$strings = $this->parsers->reduce( 
   function( $strings, $parser ) use ( $block ) {
      return $strings->merge( $parser->find( $block ) ); 
  },
  collect( [] )
)->toArray();
```

Reply:
The working code is:

```php
// With $parsers collected in the constructor.
$strings = $this->parsers->reduce(
	function( $strings, $parser ) use ( $block ) {
		return collect( $strings )->merge( collect( $parser->find( $block ) ) );
	},
	collect( [] )
)->toArray();
```

To me it seems harder to read than:

```php
foreach ( $this->parsers as $parser ) {
	$strings = array_merge( $strings, $parser->find( $block ) );
}
```

Maybe because I am not so used to this paradigm.

Reply:
Yes, keep the simpler version. But I'm surprised you needed those extra collect because `$strings` in the function should have already been a collection.

Reply:
Well, if we really implement the composite pattern, the `HTML` and `Attributes` classes should have the same method signatures as `Collection`. For `find` we are supposed to return an `array`.